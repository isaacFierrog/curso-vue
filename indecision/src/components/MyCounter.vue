<template>
  <section>
    <h3>Square: {{ squareCounter }}</h3>
    <h3>Counter: {{ counter }}</h3>

    <div>
      <button @click="incrementar">+1</button>
      <button @click="decrementar">-1</button>
    </div>
  </section>
</template>

<script lang="ts" setup>
import { useCounter } from '@/composables/useCounter';

interface Props {
  value: number;
}
defineProps<Props>();

const { counter, squareCounter, incrementar, decrementar } = useCounter(10);
</script>
