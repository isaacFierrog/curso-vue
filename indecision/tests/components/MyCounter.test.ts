import { mount } from '@vue/test-utils'
import MyCounter from '@/components/MyCounter.vue'

describe('<MyCounter />', () => {
    test('should match snapchot', () => {
        const wrapper = mount(MyCounter, {
            props: {
                value: 5
            },
        });

        expect(wrapper.html()).toMatchSnapshot()
    })

    test('renders the counter value correctly', () => {
        const wrapper = mount(MyCounter, {
            props: {
                value: 5
            },
        });

        console.log(wrapper.find('h3'));

        console.log(wrapper.html())
    })
})