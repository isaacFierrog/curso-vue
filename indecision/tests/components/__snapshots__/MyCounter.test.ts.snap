// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<MyCounter /> > should match snapchot 1`] = `
"<section>
  <h3>Square: 100</h3>
  <h3>Counter: 10</h3>
  <div><button>+1</button><button>-1</button></div>
</section>"
`;
